mypy/plugins/attrs.py

```diff
@@ -319,6 +328,51 @@ def attr_class_maker_callback(ctx: 'mypy.plugin.ClassDefContext',
     if frozen:
         _make_frozen(ctx, attributes)
 
+    # Set up the `__attrs_attrs__` class variable.
+    attr_type: TypeInfo = lookup_fully_qualified("attr.Attribute",
+        ctx.api.modules,
+        raise_on_missing=True).node
+    tuple_type: TypeInfo = lookup_fully_qualified("builtins.tuple",
+        ctx.api.modules,
+        raise_on_missing=True).node
+
+    sym_table = SymbolTable(
+        {
+            a.name: SymbolTableNode(
+                MDEF,
+                _make_var(
+                    a.name,
+                    Instance(attr_type,
+                        # We want to support future attrs versions with a 2-param Attribute
+                        [Instance(ctx.cls.info, []), a.argument(ctx).type_annotation]
+                        if len(attr_type.type_vars) == 2 else
+                        [a.argument(ctx).type_annotation]),
+                ),
+            )
+            for a in attributes
+        },
+    )
+    cd = ClassDef(f"{info.name}Attributes", Block([]))
+    cd.fullname = f"{info.name}Attributes"
```

Comment:
Shouldn't it be `f"{info.fullname}Attributes"`? https://github.com/python/mypy/blob/master/mypy/nodes.py#L2471

Reply:
Looking at the runtime behavior, looks like the inner class is actually just called `{name}Attributes`, and the `__module__` is 'builtins', always.

```python3
>>> import a
>>> a.A
<class 'a.A'>
>>> a.A.__attrs_attrs__
(Attribute(name='x', default=NOTHING, validator=None, repr=True, eq=True, eq_key=None, order=True, order_key=None, hash=None, init=True, metadata=mappingproxy({}), type=<class 'str'>, converter=<function converter at 0x7fcd03db2040>, kw_only=False, inherited=False, on_setattr=None),)
>>> a.A.__attrs_attrs__.__class__
<class 'AAttributes'>
>>> a.A.__attrs_attrs__.__module__
'builtins'
>>>
```